src/capture.rs

```diff
@@ -2,33 +2,36 @@ use std::collections::HashSet;
 use std::ops::Deref;
 use std::sync::Arc;
 
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use bytes::Bytes;
 
 use axum::{http::StatusCode, Json};
 // TODO: stream this instead
 use axum::extract::{Query, State};
 use axum::http::HeaderMap;
+use axum_client_ip::InsecureClientIp;
 use base64::Engine;
-use uuid::Uuid;
-
-use crate::api::CaptureResponseCode;
-use crate::event::ProcessedEvent;
+use serde_json::Value;
 
 use crate::{
-    api::CaptureResponse,
-    event::{Event, EventFormData, EventQuery},
+    api::{CaptureResponse, CaptureResponseCode},
+    event::{EventFormData, EventQuery, ProcessedEvent, RawEvent},
     router, sink, token,
+    utils::uuid_v7,
 };
 
 pub async fn event(
     state: State<router::State>,
-    meta: Query<EventQuery>,
+    InsecureClientIp(ip): InsecureClientIp,
+    mut meta: Query<EventQuery>,
     headers: HeaderMap,
     body: Bytes,
 ) -> Result<Json<CaptureResponse>, (StatusCode, String)> {
     tracing::debug!(len = body.len(), "new event request");
 
+    meta.now = Some(state.timesource.current_time());
```

Comment:
Do these ever get sent up by the client? If not, might be best to pass them separately from `meta`, and avoid the `mut`